EAP7-881.adoc

```diff
@@ -0,0 +1,88 @@
+= Provide launch.openshift.io experience
+:author:            Emmanuel Hugonnet
+:email:             ehugonne@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          openshift,launch,developer
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-881
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+This support is provided by Fabric8 launcher: https://github.com/fabric8-launcher.
+
+=== Other Interested Projects
+
+== Requirements
+
+ * provide generators for creation of new applications with WildFly
+ * provide generators for creation of new applications with JBoss EAP
+ * provide a way to discriminate WildFly / JBoss EAP applications
+ * provide _missions_ for the deployment of example applications (trying to reuse our Quickstarts code).
+
+
+== Implementation Plan
+
+There is absolutely NO front-end changes involved, that means that we can concentrate on the backend services of the launcher.
+
+=== Create a New Application
+
+We need to provide the following capabilities of the launcher creator:
+
+- database: to support connection and persistence on a remote database (currently MySQL and Postgresql are supported)
+- health: to provide an health check endpoint.
+- rest: to provide a REST HelloWorld API.
+ 
+Following the document https://github.com/fabric8-launcher/launcher-creator-backend/wiki/Contributing  we are going to provide the generators to support those capabilities:
+
+* create a *platform-wildfly* generator to define the base configuration of a WildFly application with a health check endpoint on _/health_ returning a simple OK.
+Since this must be exposed we can't use the Microprofile Health support of WildFly which is on 9990.
```

Comment:
What is this then? Some custom user created webapp that provides their own 'health' context? Why would we teach people to do that?

Reply:
This is used by the launcher UI